### ISSUE TITLE
Expected more than 1 value per channel when training, got input size torch.Size([1, **])

### ISSUE THREAD

**Issue by iamstarlee** [iamstarlee/4_Customized_Image_Classifier#2]
what does [BatchNorm1d](https://pytorch.org/docs/stable/generated/torch.nn.BatchNorm1d.html#torch.nn.BatchNorm1d)  do mathematically?
try and write down the equation for the case of batch_size=1 and you'll understand why PyTorch is angry with you.

How to solve it?
It is simple: BatchNorm has two "modes of operation": one is for training where it estimates the current batch's mean and variance (this is why you must have batch_size>1 for training).
 The other "mode" is for evaluation: it uses accumulated mean and variance to normalize new inputs without re-estimating the mean and variance. In this mode, there are no problems processing samples one by one.

When evaluating your model use [model.eval()](https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module.eval) before and [model.train()](https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module.train) after.

When the model is in its "training phase" it should be in model.train() state, when evaluating/testing the model it should be in model.eval() state. In your code, these two phases are a little mixed in the main loop. But basically, the code in that loop under torch.no_grad() is an evaluation code, you should have model.eval() at the beginning and switch back to model.train() after it

**Comment by iamstarlee** [comment]
write down some issues when I run the test code
